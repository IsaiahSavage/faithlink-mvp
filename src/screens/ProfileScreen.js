import React from 'react';
import { View, Text, StyleSheet, FlatList } from 'react-native';
import LinkButton from '../components/LinkButton';

const LINKS = [
  {
    screen: 'Settings',
    text: 'Settings',
  },
  {
    screen: 'Redirect',
    text: 'Scan Into Chapel',
  },
  {
    screen: 'TermsOfUse',
    text: 'Terms of Use',
  },
  {
    screen: 'PrivacyPolicy',
    text: 'Privacy Policy',
  },
  {
    screen: 'ContactSupport',
    text: 'Contact Support',
  },
];

const ProfileScreen = () => {
  return (
    <View style={styles.contentContainer}>
      <View style={styles.loginStatsContainer}>
        <Text style={styles.loginStatsStreak}>Streak: 128</Text>
        <Text style={styles.loginStatsWeek}>This week: 6</Text>
      </View>
      <View style={styles.settingsContainer}>
        {/* 
        // This is in case we switch back to a 2-column layout with smaller button option
        <FlatList
          data={LINKS.filter((e) => e.size === 'small')}
          renderItem={({ item }) => (
            <LinkButton
              to={{ screen: `${item.screen}` }}
              containerStyles={[
                styles.settingsButton,
                styles.settingsButtonSmall,
              ]}
              textStyles={[
                styles.settingsButtonText,
                styles.settingsButtonSmallText,
              ]}
            >
              {item.text}
            </LinkButton>
          )}
          keyExtractor={(item) => `${item.screen}`}
          numColumns={2}
          columnWrapperStyle={{ justifyContent: 'space-between' }}
          scrollEnabled={false}
          style={[styles.settingsGroupContainer]}
        /> */}
        <FlatList
          data={LINKS}
          renderItem={({ item }) => (
            <LinkButton
              to={{ screen: `${item.screen}` }}
              containerStyles={styles.settingsButton}
              textStyles={styles.settingsButtonText}
            >
              {item.text}
            </LinkButton>
          )}
          keyExtractor={(item) => item.screen}
          scrollEnabled={false}
          style={styles.settingsGroupContainer}
        />
      </View>
      <Text style={styles.deviceInfo}>
        Device ID: 123456789, Release Version v0.0.1a
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  contentContainer: {
    flex: 1,
    height: '100%',
  },
  loginStatsContainer: {
    flexDirection: 'row',
    marginHorizontal: 10,
    marginTop: 25,
  },
  loginStatsStreak: {
    color: '#002857',
    fontSize: 20,
    marginRight: 10,
  },
  loginStatsWeek: {
    color: '#337AB7',
    fontSize: 20,
  },
  settingsContainer: {
    marginTop: 20,
  },
  settingsGroupContainer: {
    marginHorizontal: 10,
  },
  settingsButton: {
    backgroundColor: '#E8E8E8',
    borderRadius: 100,
    justifyContent: 'center',
    paddingVertical: 16,
    paddingHorizontal: 32,
    marginBottom: 15,
  },
  settingsButtonText: {
    color: '#337AB7',
    fontSize: 18,
    fontWeight: 'bold',
  },
  // See comment above about smaller button option
  // settingsButtonSmall: {
  //   flex: 0.5,
  //   width: '50%',
  //   marginHorizontal: 10,
  // },
  // settingsButtonSmallText: {
  //   textAlign: 'center',
  // },
  deviceInfo: {
    color: '#E8E8E8',
    textAlign: 'center',
  },
});

export default ProfileScreen;
