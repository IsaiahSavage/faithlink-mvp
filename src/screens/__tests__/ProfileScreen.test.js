import React from 'react';
import { render } from '@testing-library/react-native';
import ProfileScreen from '../ProfileScreen';

jest.mock('../../contexts/UserContext.js', () => ({
  __esModule: true,
  useUserContext: jest.fn(() => ({
    state: {
      // Mocked user data
      userID: 1,
      userData: {
        id: 1,
        first: 'John',
        last: 'Doe',
        email: 'john.doe@example.com',
      },
    },
    // Mocked function
    dispatch: jest.fn(),
  })),
}));

jest.mock('../../components/LoginTracker.js', () => {
  return function MockLoginTracker() {
    const React = require('react');
    const { View } = require('react-native');
    return <View testID="loginTracker"></View>;
  };
});

jest.mock('../../components/HabitTracker.js', () => {
  return function MockHabitTracker() {
    const React = require('react');
    const { View } = require('react-native');
    return <View testID="habitTracker"></View>;
  };
});

describe('ProfileScreen', () => {
  it('renders without crashing', () => {
    render(<ProfileScreen />);
  });

  it('displays the welcome message', () => {
    const { getByText } = render(<ProfileScreen />);
    const welcomeMessage = getByText(/Welcome, John!/);
    expect(welcomeMessage).toBeTruthy();
  });

  it('displays the login tracker component', () => {
    const { getByTestId } = render(<ProfileScreen />);
    const loginTracker = getByTestId('loginTracker');
    expect(loginTracker).toBeTruthy();
  });

  it('displays the habit tracker component', () => {
    const { getByTestId } = render(<ProfileScreen />);
    const habitTracker = getByTestId('habitTracker');
    expect(habitTracker).toBeTruthy();
  });
});
