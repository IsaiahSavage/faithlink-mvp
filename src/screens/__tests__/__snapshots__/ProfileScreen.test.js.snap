// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileScreen matches the snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      style={
        {
          "backgroundColor": "white",
          "flex": 1,
        }
      }
    >
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "marginTop": 10,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-evenly",
                "marginTop": 10,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#002857",
                  "flexBasis": "50%",
                  "fontSize": 40,
                  "fontWeight": "bold",
                  "marginBottom": 25,
                  "marginLeft": 10,
                }
              }
            >
              Welcome,
               
              John
              !
            </Text>
            <View
              testID="loginTracker"
            />
          </View>
          <View
            testID="habitTracker"
          />
        </View>
      </View>
    </RCTScrollView>
  </View>
</RNCSafeAreaProvider>
`;
